File: base-java/src/main/java/tech/thdev/base/presenter/CommonPresenter.java
package tech.thdev.base.presenter;

/**
 * Created by tae-hwan on 5/6/17.
 */

public abstract class CommonPresenter<VIEW extends BaseView> implements BasePresenter<VIEW> {

    protected VIEW view;

    @Override
    public void attachView(VIEW view) {
        this.view = view;
    }

    @Override
    public void detachView() {
        view = null;
    }

    protected boolean isDetached() {
        return view != null && view.isNotFinish();
    }
}

File: base-kotlin/src/main/java/tech/thdev/base/presenter/CommonPresenter.kt
package tech.thdev.base.presenter

/**
 * Created by tae-hwan on 8/17/16.
 */
abstract class CommonPresenter<VIEW : BaseView> : BasePresenter<VIEW> {

    protected var view: VIEW? = null
        private set


    override fun attachView(view: VIEW) {
        this.view = view
    }

    override fun detachView() {
        view = null
    }
}